starlark/testdata/bench_repeat.star

```diff
@@ -0,0 +1,13 @@
+# Benchmark repetition of a tuple/list
```

Comment:
```
File/bench_repeat.star/repeat_empty_tuple-8       4.18s ± 1%   0.00s ± 0%  -100.00%  (p=0.000 n=10+9)
File/bench_repeat.star/repeat_short_tuple_10-8    567ns ± 0%   523ns ± 0%    -7.68%  (p=0.000 n=10+8)
File/bench_repeat.star/repeat_short_tuple_10k-8   255µs ± 0%   158µs ± 0%   -38.25%  (p=0.000 n=9+10)
File/bench_repeat.star/repeat_short_tuple_10m-8   594ms ± 1%   510ms ± 5%   -14.19%  (p=0.000 n=9+9)
```

What does the 4.18s figure mean? Even without the n*len(elems)==0 shortcut it can't have been that slow to allocate and copy zero bytes.

I like the changes to the test but I am wary of adding a benchmark for something that probably doesn't matter to any real program. The new implementation of repeat, but that is both a plus and a minus: the existing implementation is asymptotically optimal and less than 2x slower in the worst case of repeating a short list many times, which I suspect is highly atypical.  For smaller repeats the difference is marginal and for larger repeats the cost of allocation dominates.

I don't want to make a big deal out of a small thing but equally I don't want to set a precedent of overoptimizing at the expense of readability or of being led astray by the existence of benchmarks that don't matter. Which is to say, we really do need more realistic benchmarks.